divviup/src/main/java/org/divviup/android/TaskId.java

```diff
@@ -0,0 +1,33 @@
+package org.divviup.android;
+
+import android.util.Base64;
+
+public class TaskId {
+    private final byte[] bytes;
+
+    public TaskId(byte[] bytes) {
```

Comment:
`TaskId` is not effectively immutable: if a caller passed in a given `bytes` array, then later modified the array, those modifications would show up in `TaskId`'s byte array (since they're the same byte array).

Given that this byte array is ultimately passed down to the JNI'ed Rust code directly, and the Rust code's safety guarantees require that there be no concurrent modifications to the byte arrays that are passed to it, I think we should fix this. Otherwise we could have UB if someone uses this library incorrectly.

We should make constructing a `TaskId` based on an arbitrary byte array private, and ensure that all public constructors/static-factory-methods effectively create a `TaskId` based on a `byte[]` that nothing else could possibly have a reference to (e.g. by making defensive copies as-needed).

But even that isn't quite good enough: `toBytes()` returns a reference to the `TaskId`'s byte array, which can be modified. We could reduce visibility on `toBytes()`--I suppose only the library implementation itself needs to be able to get this byte array, users of the library can use `parse()` and `encodeToString()`. If we do reduce visibility, I still think it's important to document that the byte array shouldn't be modified since modifying it could lead to UB. (The other option that comes to mind would be to make a copy with each call to `toBytes()`, but that's a bit wasteful.)